### DISCUSSION TITLE
Using Supabase `createClient` in nextjs API routes

### DISCUSSION THREAD

**Discussion by apetta** [supabase/supabase#7866] [Questions]
Hi All,

I'm trying to make use of the Supabase JS library (to access the database) via NextJS API routes. I am using a third-party auth solution, so I cannot connect from the user/browser as I'm not using RLS and need to apply additional logic before allowing read/write operations, hence the API routes as a 'middle' layer. 

Is it okay to use 'createClient' from within the API route like this? i.e.

```
// /api/getSomeData

import { createClient } from "@supabase/supabase-js";

const supabase = createClient(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_ANON_KEY
);

export default async function handler(req, res) {

  // custom auth and read/write checks

  let { data, error } = await supabase.from("someTable").select('*');

  if (error) {
    return res.status(500).json({ error });
  }

  res.status(200).json(data);
}
```

or would this cause issues at scale as it might be creating a new connection on every API request? Would appreciate some input on the best way to approach this.

**Comment by GaryAustin1** [comment]
You might want to look at https://github.com/supabase-community/auth-helpers
Either for a solution, or for ideas.  There is a next.js section.
